### DISCUSSION TITLE
Add predefined regexes and autogenerated string fields.

### DISCUSSION THREAD

**Discussion by lyneal** [directus/directus#13950] [Feature Requests]
Add predefined regexes. 

As far as I know, there are no email or phone fields.
I guess the only way is to make a regex to validate if the field is an email or a phone number, or to create extensions (interface) of the desired type. 

The idea is to propose a list of regexes
- either in the part [validation -> matches regExp -> (list of expressions)].
- or in the part [schema -> select (list of expressions)].

**Edited 10/18/22:**

I recently did some research and wondered about the possibilities of going even further with regexes (maybe a new discussion should be opened on this topic)

Here are two interesting libraries:

- Pomsky https://github.com/rulex-rs/pomsky 
Pomsky allows to facilitate the use of regexes thanks to the possibility to use variables, comments and a simplified syntax.
- Randexp https://github.com/fent/randexp.js
Randexp allows to generate a random string according to a given regex.

In addition to allowing reusability of regexes we could allow to generate strings from a regex.
This could allow to create autogenerated fields.

**Comment by benhaynes** [comment]
I think this is a great idea for validation... 👍

**Comment by sxsws** [comment]
I have been thinking about this actually and was wondering about pre-defined field types that are 'Email Field' / 'URL Field' etc that handle all this automagically. There must be existing libraries that the validation can be outsourced to, rather than using a regex.

**Reply by lyneal** [reply]
Indeed, my examples are not necessarily the best ones, but in other cases, it can be interesting, especially in terms of reusability.

_I have updated the idea._

**Comment by thinktrans** [comment]
IMO, the regex paradigm for validation of the emails and domain names is not wise from the POV of the Internationalized Domain Names and Internationalized Email IDs. 
I am happy to point to more information if the community so desires. :)

**Comment by Michael-Nussbaumer** [comment]
I had the same/similar idea. It would be great if you can also predefine some validations which can simple be reused in other collections/fields.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
